test/test_queries.py

```diff
@@ -0,0 +1,68 @@
+'''
+test_queries.py
+Contains test cases for database query related functions.
+'''
+from nose.tools import assert_equal
+from components import model
+
+# HOW TO RUN NOSE TESTS
+# 1. Make sure you are in cs-modify main directory
+# 2. Make sure the path "C:\Python27\Scripts" is added in your environment variables
+# 3. Enter in cmd: "nosetests test/"
+# 4. Nose will run all the tests inside the test/ folder
+
+class TestCode(object):
+    '''
+        This class runs the test cases for database query related functions.
+    '''
+    def __init__(self):
+        self.num_column_for_each_year = 2
+        self.num_column_for_each_focus = 2
+
+    def test_query_num_students_in_year_of_study(self):
+        '''
+            Tests querying number of students at each year of study
+        '''
+        num_in_year = [4, 3, 3, 3]
+
+        table_of_year_of_study_with_count = \
+            model.get_num_students_by_yr_study()
+
+        num_of_rows_expected = 4
+        for index_row in range(1, num_of_rows_expected):
```

Comment:
For the first part, actually I missed out year 1 students since it should be 0-based indexing. Will change to range(0, 4) instead.

For the second part, this code is used to test the current database which currently only have year 1 to year 4 students and I can't test beyond that because year 5 or more does not exist in the current database. As for the implementation code at model.py, rest assured that it can handle any number of years with no maximum status, so the implementation is safe.